tests/integ/modin/frame/test_align.py

```diff
@@ -0,0 +1,214 @@
+
+
+@sql_count_checker(query_count=0)
+def test_align_frame_fill_value_negative():
+    df = pd.DataFrame([[1, 2, 3, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"])
+    other_df = pd.DataFrame(
+        [[10, 20, 30, 40], [60, 70, 80, 90], [600, 700, 800, 900]],
+        columns=["A", "B", "C", "D"],
+    )
+    with pytest.raises(
+        NotImplementedError,
+        match="Snowpark pandas 'align' method doesn't support 'fill_value'",
+    ):
+        left, right = df.align(other_df, join="outer", axis=0, fill_value="empty")
+
+
+@sql_count_checker(query_count=0)
+def test_align_frame_axis1_negative():
+    df = pd.DataFrame(
+        [[1, 2, 3, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"], index=[1, 2]
+    )
+    other_df = pd.DataFrame(
+        [[10, 20, 30, 40], [60, 70, 80, 90], [600, 700, 800, 900]],
+        columns=["A", "B", "C", "D"],
+        index=[2, 3, 4],
+    )
+    with pytest.raises(
+        NotImplementedError,
+        match="Snowpark pandas 'align' method doesn't support 'axis=1'",
+    ):
+        left, right = df.align(other_df, join="outer", axis=1)
+
+
+@sql_count_checker(query_count=0)
+@pytest.mark.parametrize("level", [0, 1])
+def test_multiindex_negative(level):
+    df = pd.DataFrame(
+        [[1], [2]],
+        index=pd.MultiIndex.from_tuples(
+            [("foo", "bah", "ack"), ("bar", "bas", "bar")], names=["a", "b", "c"]
+        ),
+        columns=["num"],
+    )
+    other_df = pd.DataFrame(
+        [[2], [3]],
+        index=pd.Series(["foo", "bah"], name="a"),
+        columns=["num"],
+    )
+    with pytest.raises(
+        NotImplementedError,
+        match="Snowpark pandas 'align' method doesn't support 'level'",
+    ):
+        left, right = df.align(other_df, join="outer", axis=0, level=0)
+
+
+@sql_count_checker(query_count=0)
+def test_align_frame_copy_negative():
+    df = pd.DataFrame([[1, 2, 3, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"])
+    other_df = pd.DataFrame(
+        [[10, 20, 30, 40], [60, 70, 80, 90], [600, 700, 800, 900]],
+        columns=["A", "B", "C", "D"],
+    )
+    with pytest.raises(
+        NotImplementedError,
+        match="Snowpark pandas 'align' method doesn't support 'copy=False'",
+    ):
+        left, right = df.align(other_df, join="outer", axis=0, copy=False)
+
+
+@sql_count_checker(query_count=0)
+def test_align_frame_invalid_axis_negative():
+    df = pd.DataFrame([[1, 2, 3, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"])
+    other_df = pd.DataFrame(
+        [[10, 20, 30, 40], [60, 70, 80, 90], [600, 700, 800, 900]],
+        columns=["A", "B", "C", "D"],
+    )
+    axis = 2
+    with pytest.raises(
+        ValueError,
+        match=f"No axis named {axis} for object type DataFrame",
+    ):
+        left, right = df.align(other_df, join="outer", axis=axis)
+
+
+@sql_count_checker(query_count=0)
+@pytest.mark.parametrize("method", ["backfill", "bfill", "pad", "ffill"])
+def test_align_frame_deprecated_negative(method):
```

Comment:
maybe we should separate the method parametrization from the other tests? Otherwise the 4 other NotImplementedError parts get run 3 extra times.

Reply:
Good point, done!